### PR TITLE
Modify ShallowConvNet and update USPS data preprocessing

### PR DESCRIPTION
**Pull request by YanisLalou**
1. Refactored the `ShallowConvNet` class

2. Updated the USPS data preprocessing in the `_download_data` method:
   - Updated the normalization statistics for the USPS dataset from `(0.1307, 0.3081)` to `(0.0806, 0.2063)`.

3. Adjusted the default hyperparameters in the `DeepCORAL` class

4. Updated the `get_estimator` method in the `DeepCORAL ` class to use the Adadelta optimizer instead of AdamW.

### CODE REVIEW

File: datasets/mnist_usps.py [hunk 1]
```diff
@@ -118,8 +118,15 @@ def get_data(self):
...
-        X_target, y_target = self._get_dataset(data, target, self.n_samples_target)
+        X_source, y_source = self._get_dataset(
+            data, source, self.n_samples_source)
+        X_target, y_target = self._get_dataset(
+            data, target, self.n_samples_target)
+
+        print(f'Mnist mean {X_source.mean()}')
+        print(f'Mnist std {X_source.std()}')
+        print(f'USPS mean {X_target.mean()}')
+        print(f'USPS std {X_target.std()}')
```
[the remark this answers is not in the document]
Comment (YanisLalou) [reply datasets/mnist_usps.py:129]: Yes:
```
Mnist mean 6.244002724997699e-05
Mnist std 1.0001322031021118
USPS mean -0.00012829367187805474
USPS std 1.0000249147415161
```

**Comment by antoinecollas** [comment]
Generated by @YanisLalou using this PR:
[output_readable_dataset_target.csv](https://github.com/user-attachments/files/16935783/output_readable_dataset_target.csv)
[output_readable_dataset_source.csv](https://github.com/user-attachments/files/16935784/output_readable_dataset_source.csv)

**Comment by antoinecollas** [comment]
From table 2 of https://arxiv.org/pdf/1803.10081:
- MNIST → USPS is comparable (~94%)
- we do better on USPS → MNIST (73% vs 60%) 
- MNIST -> MNIST is comparable (~99%)
- USPS -> USPS, we do better (97.5% vs 96%)

File: solvers/DeepCORAL.py [hunk 2]
```diff
@@ -48,14 +48,23 @@ def get_estimator(self, n_classes, device, dataset_name, **kwargs):
...
+        
+        
         net = DeepCoral(
             model,
-            optimizer=AdamW,
-            layer_name="feature_layer",
+            optimizer=Adadelta,
+            layer_name="fc1",
             batch_size=256,
             max_epochs=1,
```
Comment (antoinecollas) [inline solvers/DeepCORAL.py:64]: Two minor comments:
- by default in https://github.com/pytorch/examples/blob/main/mnist/main.py#L75 :  batch_size=64 (but bigger is better)
- you set `max_epochs=1` , I guess it is overwritten by the `param_grid` (i.e. `max_epochs=14`)
